### ISSUE TITLE
Support payjoin for automatic timelock refresh

### ISSUE THREAD

**Issue by DanGould** [wizardsardine/liana#534]
@edouardparis mentioned receiving [payjoin](https://payjoin.org) for automatic relative timelock [refresh](https://github.com/wizardsardine/liana/blob/0406085ed5a23bfcfb94eb36ac79a2734ae38a3d/README.md?plain=1#L88-L970) at Bitcoin 2023. Since payjoin enables output substitution during receipt, timelocked coins can be spent into a fresh timelock at the same time as incoming payment. The sender even subsidies the transaction fees. the [`rust-payjoin`](https://github.com/payjoin/rust-payjoin) crate may facilitate this upgrade in Liana. A relay server as in [serverless payjoin](https://gist.github.com/DanGould/243e418752fff760c9f6b23bba8a32f9) may be used to enable this feature without requiring the end-user to run Tor or host a server.

**Comment by Sami32** [comment]
https://payjoin.org/docs/how-it-works/payjoin-v2-bip-77/

**Comment by DanGould** [comment]
Is this something @wizardsardine might be interested in opening for work or potentially offering support for hackers at the MIT Bitcoin Expo Hackathon on April 4-5, 2025, potentially with extra support from HRF? Would a Proof of Concept solving this issue in this repo from an outside contributor be more useful than the review burden it might incur?

**Comment by kloaec** [comment]
Hey Dan,
Potentially yes.

I believe in the current state of Liana, it would make more sense to integrate Payjoin for spending, potentially adding more UTXOs to refresh and the external UTXO(s) for privacy.
The "receive" still requires the need for signing, and that is the main thing preventing the refresh at the moment. 
It might make sense in a low velocity set up, where both participants can wait potentially days before the Tx is signed by both, but don't expect hot keys in Liana.

In the future, I would hope to have a "hot key" hsm signing only refresh type transactions, and that could make the receive payjoin very useful for refresh too. No need for the user to sign from their own HW. (we aren't there yet!)

TLDR:
- a Payjoin mode for sending might make more sense
- a Payjoin mode for receiving might be somewhat limited to people trying to optimize refreshes, maybe even between Liana users, but with potentially multiple days before the tx to be signed

Happy to have a chat, drop me an email :)

The team might add their own considerations here, i'm talking for myself.

**Comment by pythcoiner** [comment]
> Would a Proof of Concept solving this issue in this repo from an outside contributor be more useful than the review burden it might incur?

from my POV something that can be useful could be a PoC that:
 - release the actual contraint the wallet have on inputs ownership (iirc Liana do not accept psbt if inputs are not "owned")
 - potentially expand the RPC API surface for the needs of payjoin (like having a specific command to import PSBTs w/ not all inputs "owned")
 - implement an external (cli) "plugin" that use the RPC to operate a painjoin

I think this should keep the review burden quite low while opening doors for a future integration of payjoin/coinjoin features
